src/Aspire.Hosting.AWS/CDK/CDKExtensions.cs

```diff
@@ -0,0 +1,252 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Amazon.CDK;
+using Amazon.CloudFormation.Model;
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.AWS;
+using Aspire.Hosting.AWS.CDK;
+using Aspire.Hosting.AWS.CloudFormation;
+using Constructs;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Extensions.Logging;
+using Stack = Amazon.CDK.Stack;
+using StackResource = Aspire.Hosting.AWS.CDK.StackResource;
+
+namespace Aspire.Hosting;
+
+/// <summary>
+/// Extension methods for adding AWS CDK as a provisioning resources.
+/// </summary>
+public static class CDKExtensions
+{
+    /// <summary>
+    ///
+    /// </summary>
+    /// <param name="builder"></param>
+    /// <param name="name"></param>
+    /// <param name="props"></param>
+    /// <returns></returns>
+    public static IResourceBuilder<IAppResource> AddAWSCDK(this IDistributedApplicationBuilder builder, string name, IAppProps? props = null)
+    {
+        builder.Services.TryAddSingleton<ICloudFormationProvisionerFactory, CDKProvisionerFactory>();
```

Comment:
The factory pattern here seems problematic because if I start the AppHost with adding a CloudFormation template using the a template file and then later add some CDK resources the implementation of `ICloudFormationProvisionerFactory` will be `CloudFormationProvisionerFactory` not `CDKProvisionerFactory` so the CDK deployment won't work correctly.

If a user has both a template file and CDK constructs that should play nice with each other. I would be fine with putting all of the CDK work in `Aspire.Hosting.AWS.CloudFormation.CDK` namespace and then just have one provisioner that knows about CloudFormation template files and CDK constructs. I don't think we need to treat "raw" CloudFormation and CDK as sibling features.

Reply:
Yeah, maybe it is easier as we have put everything in on project anyway. This is now a different features, but I agree that you might need the flexibility to use both.